gdb_runner.py

```diff
@@ -0,0 +1,224 @@
+# This file is part of the Test Environment build system.
+#
+# @copyright 2020-2021 N7 Space Sp. z o.o.
+#
+# Test Environment was developed under a programme of,
+# and funded by, the European Space Agency (the "ESA").
+#
+#
+# Licensed under the ESA Public License (ESA-PL) Permissive,
+# Version 2.3 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://essr.esa.int/license/list
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import binascii
+import atexit
+import configparser
+import time
+import timeout_decorator
+import os
+
+from libs.GdbInterface import GdbInterface
+from libs.GdbServerInvoker import GdbServerInvoker
+from libs.ConnectionConfig import ConnectionConfig
+
+from configparser import ConfigParser
+
+class gdb_runner:
+    __gdbSrv = None
+    __gdb = None
+    __ioConsole = None
+    __ioUart4 = None
+
+    __config = ConfigParser()
+
+    def __init__(self, configPath, vConsole, vUart4):
+        self.__config = ConfigParser(os.environ)
+        self.__config.read(configPath)
+        self.__config.set('ioConsole', 'vPortName', vConsole)
+        self.__config.set('ioUart4', 'vPortName', vUart4)
+
+    def __log(msg, *args, **kwargs):
+        print("GDB-RUNNER:", msg, *args, **kwargs)
+
+    def __invokeGdbServer(self):
+        self.__log("Starting GDB server...")
+        srv = GdbServerInvoker(
+            self.__config.get('gdbServer', 'path'),
+            self.__config.get('gdbServer', 'args'),
+            ConnectionConfig.fromConfig(self.__config['gdbServer']),
+            verbosity=True,
+        )
+        srv.open()
+        time.sleep(1)
+        return srv
+
+
+    def __invokeGdb(self):
+        self.__log("Staring GDB...")
+        gdb = GdbInterface(
+            self.__config.get('gdb' ,'address'),
+            self.__config.get('gdb' ,'path'),
+            verbose=self.__config.get('gdb' ,'verbose'),
+        )
+        gdb.launch()
+        return gdb
+
+
+    def __openIoHandler(self, config):
+        self.__log("Starting IO Handler...")
+
+        from libs.UartIoHandler import UartIoHandler
+        handler = UartIoHandler(
+                config["address"],
+                config["username"],
+                config["password"],
+                config["path"],
+                config["baudrate"],
+                config["port"],
+                config["vPortName"],
+                debug=config["verbose"],
+            )
+        handler.open()
+        return handler
+
+
+    def __dumpLogLambda(self, io, logPath):
```

Comment:
What is lambda?

Reply:
OK... I see... but I don't think this is really a lambda. I would rename it to e.g __dumpLogInternal or better __receiveAndWriteLog.